src/strategies/spirit/index.ts

```diff
@@ -0,0 +1,99 @@
+import { multicall } from '../../utils';
+import { formatUnits } from '@ethersproject/units';
+
+export const author = 'chimpydev';
+export const version = '0.1.0';
+
+async function search_contract_cretion_block(contract_address, provider) {
+  var highest_block = await provider.getBlockNumber();
+  var lowest_block = 13691035;
+
+  var contract_code = await provider.getCode(contract_address, highest_block);
+
+  if (contract_code == '0x') {
+    return -1;
+  }
+
+  while (lowest_block <= highest_block) {
+    let search_block = Math.floor((lowest_block + highest_block) / 2);
+    contract_code = await provider.getCode(contract_address, search_block);
+    if (contract_code != '0x') {
```

Comment:
There is limit of 5 requests being sent from a strategy, this looks like this can send many requests to find exact block

https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Can you find someother alternate way to do this?

Reply:
I have writed about this on the READ.ME file.
So this is like contract-call strategy, you must only put one argument. 
I have spent more than 3 days on this and is the only solution that i have found